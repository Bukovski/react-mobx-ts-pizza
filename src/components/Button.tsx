import React from 'react';
import classNames from 'classnames';


interface IButton {
  onClick?: (event: React.MouseEvent<HTMLElement>) => void,
  className?: string,
  outline?: boolean,
  children: React.ReactNode,
}

const Button = (props: IButton) => {
  const { onClick, children, className, outline } = props;

  return (
    <button
      onClick={ onClick }
      className={ classNames('button', className, {
        'button--outline': outline,
      })}>
      { children }
    </button>
  );
};



export default Button;