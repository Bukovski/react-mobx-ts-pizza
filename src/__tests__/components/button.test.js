import React from 'react';
import { screen, render } from "@testing-library/react";
import Button from "../../components/button.component";
import userEvent from "@testing-library/user-event";


describe("Button Component", () => {
	let childrenText,
		mockClickFn;
	
	beforeEach(() => {
		childrenText = <span>Button text</span>;
		mockClickFn = jest.fn();
	});
	
	it("should render with children and spy on click", () => {
		const { container } = render(
			<Button
				onClick={ mockClickFn }
			>
				{ childrenText }
			</Button>
		);
		
		expect(screen.getByText("Button text")).toBeInTheDocument();
		
		expect(mockClickFn).not.toBeCalled();
		
		userEvent.click(screen.getByRole("button"));
		expect(mockClickFn).toBeCalled();
		
		expect(container.firstChild).toMatchSnapshot()
	})
	
	it("should render with custom className & outline", () => {
		const { container } = render(
			<Button
				onClick={ mockClickFn }
				className="form-order--primary"
				outline
			>
				{ childrenText }
			</Button>
		);
		
		const buttonVar = screen.getByRole("button");
		
		expect(buttonVar).toHaveClass("form-order--primary");
		expect(buttonVar).toHaveClass("button--outline");
		
		expect(container.firstChild).toMatchSnapshot()
	})
});
