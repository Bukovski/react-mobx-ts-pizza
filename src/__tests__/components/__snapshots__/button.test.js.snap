// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render with children and spy on click 1`] = `
<button
  class="button"
>
  <span>
    Button text
  </span>
</button>
`;

exports[`Button Component should render with custom className & outline 1`] = `
<button
  class="button form-order--primary button--outline"
>
  <span>
    Button text
  </span>
</button>
`;
